test/node/double_tests.js

```diff
@@ -3,36 +3,96 @@
 const BSON = require('../register-bson');
 const Double = BSON.Double;
 
-describe('Double', function () {
-  describe('Constructor', function () {
-    var value = 42.3456;
+describe('BSON Double Precision', function () {
+  context('class Double', function () {
+    describe('constructor()', function () {
+      const value = 42.3456;
 
-    it('Primitive number', function (done) {
-      expect(new Double(value).valueOf()).to.equal(value);
-      done();
+      it('Primitive number', function (done) {
+        expect(new Double(value).valueOf()).to.equal(value);
+        done();
+      });
+
+      it('Number object', function (done) {
+        expect(new Double(new Number(value)).valueOf()).to.equal(value);
+        done();
+      });
+    });
+
+    describe('#toString()', () => {
+      it('should serialize to a string', () => {
+        const testNumber = Math.random() * Number.MAX_VALUE;
+        const double = new Double(testNumber);
+        expect(double.toString()).to.equal(testNumber.toString());
+      });
+
+      const testRadices = [2, 8, 10, 16, 22];
+
+      for (const radix of testRadices) {
+        it(`should support radix argument: ${radix}`, () => {
+          const testNumber = Math.random() * Number.MAX_VALUE;
+          const double = new Double(testNumber);
+          expect(double.toString(radix)).to.equal(testNumber.toString(radix));
+        });
+      }
     });
+  });
+
+  function serializeThenDeserialize(value) {
+    const serializedDouble = BSON.serialize({ d: value });
+    const deserializedDouble = BSON.deserialize(serializedDouble, { promoteValues: false });
```

Comment:
what's the significance of `promoteValues: false` here? what happens with `promoteValues: true`?

Reply:
This is so we get the raw BSON instead of the value as a double.

Reply:
could you add a comment above this line explaining that? also @nbbeeken is there a more precise option we can use than the general promoteValues?

Reply:
Unfortunately Doubles do not have a dedicated promote option, promoteValues: false is the only way to get the deserializer to return instanceof Double. _chaos_